sdk/appconfiguration/app-configuration/samples/typescript/tsconfig.json

```diff
@@ -3,7 +3,10 @@
     "module": "commonjs",
     "moduleResolution": "node",
 
-    "lib": ["esnext.asynciterable"],
+    "lib": [
+      "ESNext.AsyncIterable",
+      "DOM"   // for core-http
```

Comment:
Wait...does this mean that anyone using this package with TypeScript needs to specify `DOM` in `lib` in their `tsconfig.json`? I don't think I've had to do that with libraries like storage-blob.

Reply:
Ah, I noticed you don't have a `target` set. Any reason not to set it to `ES2018` and remove the `lib` section altogether? The sample readme already says that node 8 is required, and node 10 is required for the for/await samples.

Reply:
ES2018 is too high for node 8 I think (async iterators aren't supported).

We should not need the lib entry for `dom`, if it's needed it's probably something we can fix in core-http by stubbing the appropriate types out.

Reply:
Filed issue #6821 for the core-http stubbing.

It looks like if our desire is to continue to support node 8.x then I can't go the ES2018 route, which means I am stuck with specifying `lib` and `DOM` explicitly (which also means this change would go in as-is).

Reply:
After talking with all the relevant parties we are just going to target ES2015 and remove libs altogether. :)